packages/gatsby/src/schema/node-model.js

```diff
@@ -224,6 +218,35 @@ class LocalNodeModel {
   findRootNodeAncestor(obj, predicate) {
     return this.nodeStore.findRootNodeAncestor(obj, predicate)
   }
+
+  /**
+   * Given a result, that's either a single node or an array of them, track them
+   * using pageDependencies. Defaults to tracking according to current resolver
+   * path. Returns the result back.
+   *
+   * @param {Node | Node[]} result
+   * @param {PageDependencies} [pageDependencies]
+   * @returns {Node | Node[]}
+   */
+  trackPageDependencies(result, pageDependencies) {
+    const fullDependencies = {
+      path: this.path,
+      ...(pageDependencies || {}),
+    }
+    const { path, connectionType } = fullDependencies
+    if (path) {
+      if (connectionType) {
+        this.createPageDependency({ path, connection: connectionType })
+      } else {
+        const nodes = Array.isArray(result) ? result : [result]
```

Comment:
This should work too, don't know which one is more readable. (feel free to ignore)
```suggestion
        const nodes = [].concat(result);
```